### PR TITLE
Try to fix upload-artifact@v4 issue

### PR DESCRIPTION
**Pull request by mcruzdev**
Many thanks for submitting your Pull Request :heart:!

This pull request aims to fix the issue #599.

Please make sure that your PR meets the following requirements:

- [ ] You have read the [contributors guide](CONTRIBUTING.md)
- [ ] Your code is properly formatted according to [our code style](CONTRIBUTING.md#code-style)
- [ ] Pull Request title contains the target branch if not targeting main: `[0.9.x] Subject`
- [ ] Pull Request contains link to the issue
- [ ] Pull Request contains link to any dependent or related Pull Request
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket

<details>
<summary>
How to backport a pull request to a different branch?
</summary>

In order to automatically create a **backporting pull request** please add one or more labels having the following format `backport-<branch-name>`, where `<branch-name>` is the name of the branch where the pull request must be backported to (e.g., `backport-quarkus2` to backport the original PR to the `quarkus2` branch).

> **NOTE**: **backporting** is an action aiming to move a change (usually a commit) from a branch (usually the main one) to another one, which is generally referring to a still maintained release branch. Keeping it simple: it is about to move a specific change or a set of them from one branch to another.

Once the original pull request is successfully merged, the automated action will create one backporting pull request per each label (with the previous format) that has been added.

If something goes wrong, the author will be notified and at this point a manual backporting is needed.

> **NOTE**: this automated backporting is triggered whenever a pull request on `main` branch is labeled or closed, but both conditions must be satisfied to get the new PR created.
</details>

### CODE REVIEW

**Comment by ricardozanini** [comment]
@mcruzdev  I think the variable name is not getting from the `matrix` var. See:  `Artifact site-{{ matrix.java.version }}-{{ matrix.os }}.zip successfully finalized. Artifact ID 1129900601`

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -57,7 +57,7 @@ jobs:
         if: github.event_name == 'pull_request'
         uses: actions/upload-artifact@v4
         with:
-          name: site
+          name: site-{{ matrix.java.version }}-{{ matrix.os }}
```
Comment (ricardozanini) [inline .github/workflows/build.yml:60]:
```suggestion
          name: site-${{ matrix.java.version }}-${{ matrix.os }}
```

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -96,6 +96,6 @@ jobs:
         if: github.event_name == 'pull_request'
         uses: actions/upload-artifact@v4
         with:
-          name: site
+          name: site-{{ matrix.java.version }}-{{ matrix.os }}-reactive
```
Comment (ricardozanini) [inline .github/workflows/build.yml:99]:
```suggestion
          name: site-${{ matrix.java.version }}-${{ matrix.os }}-reactive
```

File: .github/workflows/preview.yml [hunk 3]
```diff
@@ -16,7 +16,7 @@ jobs:
         with:
           workflow: ${{ github.event.workflow_run.workflow_id }}
           workflow_conclusion: success
-          name: site
+          name: site-17-ubuntu-latest
```
Comment (ricardozanini) [inline .github/workflows/preview.yml:19]: We can add a single matrix here and use the vars instead.

**Comment by ricardozanini** [comment]
Another improvement, I think we can build only once the website in the `preview` pipeline. Doesn't make much sense publishing 4 artifacts in each matrix run.

**Comment by mcruzdev** [comment]
> Another improvement, I think we can build only once the website in the `preview` pipeline. Doesn't make much sense publishing 4 artifacts in each matrix run.

$ sign mistake, TY.

**Comment by ricardozanini** [comment]
> > Another improvement, I think we can build only once the website in the `preview` pipeline. Doesn't make much sense publishing 4 artifacts in each matrix run.
> 
> $ sign mistake, TY.

Thank you for fixing it!!

**Comment by mcruzdev** [comment]
@ricardozanini WYT about creating just one job to generate the site documentation and consume it in preview.

**Comment by ricardozanini** [comment]
> @ricardozanini WYT about creating just one job to generate the site documentation and consume it in preview.

Yes, exactly. We should remove the website upload from the matrix build. The other CI can just build skipping test since it's only interested in the website generation.

**Comment by mcruzdev** [comment]
Done @ricardozanini :)

**Review by ricardozanini — APPROVED** [review]
Great work! Many thanks!!

**Comment by hbelmiro** [comment]
Awesome! Thank you very much @mcruzdev ❤️

**Comment by hbelmiro** [comment]
@all-contributors add @mcruzdev for code and documentation.

**Pull request merged**